src/Illuminate/Container/Container.php

```diff
@@ -785,6 +785,13 @@ public function build($concrete, $parameters = [])
 			throw new BindingResolutionException($message);
 		}
 
+		if (in_array($concrete, $this->buildStack))
+		{
+			$message = "Cyclic reference found while resolving [$concrete].";
+
+			throw new BindingResolutionException($message);
```

Comment:
I created a CircularReferenceException class. But I had to update the `Container::resolveClass()` method to catch the new exception.